README.md

```diff
@@ -77,7 +77,7 @@ class ValidationTask2(dvf.task.SetValidationTask):
     ValidationTask2."""
 
     def inputs(self):
-        return {ValidationTask1(): {"col_name": "new_col_name_in_current_task"}}
+        return {ValidationTask1: {"col_name": "new_col_name_in_current_task"}}
```

Comment:
I've always wondered if we don't pass a dict, but a str, if we can assume both key/values are the same? it may be to confusing, not sure

Reply:
Yeah I don't know. It could be list of str but if one want to change one column name it might be even more verbose in the end. So I don't know but if you have any idea of an API for this we can try.

Reply:
Ooooh I had an idea tonight! Maybe we could split the process: we could have a step to rename the columns of the input dataframe and then the second step imports the columns from the input tasks. So in basic case we would have this:

```python
    class MyTask(ElementValidationTask):
        validation_function = staticmethod(my_validation_function)
        output_columns = {'my_new_col': None}

        def inputs(self):
            return {
                PreviousTask: ["input_data"]
            }
```

And if we want to rename a column, we use a new mapping like this:

```python
    class MyTask(ElementValidationTask):
        validation_function = staticmethod(my_validation_function)
        output_columns = {"my_new_col": None}
        rename_columns = {"input_data": "input_data_from_dataframe"}

        def inputs(self):
            return {
                PreviousTask: ["input_data"]
            }
```

What do you think?

Reply:
Ah but there is an issue with this: we can't deal with two inputs with same column names.

Reply:
yes yes, forget it, I think it's fine as it is!

Reply:
So I am afraid we would have to to something quite confusing, like adding the column renaming feature as proposed above but still allow to rename the columns of the inputs. This could lead to something like this:

```python
    class MyTask(ElementValidationTask):
        validation_function = staticmethod(my_validation_function)
        output_columns = {"my_new_col": None}
        rename_columns = {"input_data": "input_data_from_dataframe"}

        def inputs(self):
            return {
                PreviousTaskA: ["input_data", "other_col_A"],
                PreviousTaskB: {"input_data": "input_data_from_task_B", "other_col_B": "other_col_B"},
            }
```
Which might be even more complicated in the end, though it would be simpler in simple cases.

Reply:
Ok :)